src/org/labkey/targetedms/query/SamplePivotCustomizer.java

```diff
@@ -33,92 +31,157 @@
 public class SamplePivotCustomizer implements TableCustomizer
 {
     private final FieldKey _runIdField;
+    private final String _urlParameter;
+    private final String _replicateNameQuery;
+    private final String _replicateField;
+    private final String _replicateRunIdField;
 
+    /** Referenced from query XML metadata */
+    @SuppressWarnings("unused")
     public SamplePivotCustomizer(MultiValuedMap<String, String> props)
     {
-        Collection<String> runIdFilter = props.get("runIdField");
-        if (runIdFilter == null || runIdFilter.size() != 1)
+        _runIdField = FieldKey.fromString(getParameter(props, "runIdField", true));
+        _urlParameter = getParameter(props, "urlParameter", false);
+        _replicateNameQuery = getParameter(props, "replicateNameQuery", false);
+        _replicateField = getParameter(props, "replicateField", false);
+        _replicateRunIdField = getParameter(props, "replicateRunIdField", false);
+    }
+
+    private static String getParameter(MultiValuedMap<String, String> props, String propertyName, boolean required)
+    {
+        Collection<String> values = props.get(propertyName);
+        if (values != null && values.size() == 1)
+        {
+            return values.iterator().next();
+        }
+
+        if (required)
         {
-            throw new IllegalArgumentException("Must have exactly one property named 'runIdField'");
+            throw new IllegalArgumentException("Must have exactly one property named '" + propertyName + "'");
         }
-        _runIdField = FieldKey.fromString(runIdFilter.iterator().next());
+        return null;
     }
 
-    @Override
-    public void customize(TableInfo tableInfo)
+    private Integer getRunId()
```

Comment:
As runIds are of type long now, better to be consistent across.

Reply:
We're a long way from 2 billion runs uploaded but agree best to parse as the right data type. Updated